ports/stm32/qspi.c

```diff
@@ -33,36 +33,52 @@
 
 #if defined(MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2)
 
+#ifndef MICROPY_HW_QSPI_PRESCALER
+#define MICROPY_HW_QSPI_PRESCALER       3  // F_CLK = F_AHB/3 (72MHz when CPU is 216MHz)
+#endif
+
+#ifndef MICROPY_HW_QSPI_SAMPLE_SHIFT
+#define MICROPY_HW_QSPI_SAMPLE_SHIFT    1  // sample shift enabled
+#endif
+
+#ifndef MICROPY_HW_QSPI_TIMER_COUNTER
+#define MICROPY_HW_QSPI_TIMER_COUNTER   0  // timeout counter disabled (see F7 errata)
+#endif
+
+#ifndef MICROPY_HW_QSPI_CS_HIGH_TIME
+#define MICROPY_HW_QSPI_CS_HIGH_TIME    QSPI_CS_HIGH_TIME_2_CYCLE
```

Comment:
I see you're using the HAL constant here... this is a bit inconsistent because the other 3 config values above just use integers.  So it might be better to just specify this as an integer like MICROPY_HW_QSPI_PRESCALER, then subtract 1 when it's used below so the integer does represent the number of high cycles.

Note that it doesn't actually build when using the HAL constant, at least not on STM32F769DISC (due to missing include of this HAL header).

Reply:
When I first looked at those defines I thought they were more complex than just `(n-1) << offset`, I should have looked closer...